portfolio/src/main/webapp/script.js

```diff
@@ -96,9 +96,22 @@ function showSlides(index) {
   dots[slideIndex-1].className += ' dot-active';
 }
 
-/** Fetch message from /data and push to index.html */
-async function getFruitFact() {
-  const response = await fetch('/data');
-  const message = await response.text();
-  document.getElementById('fruit-fact-container').innerText = message;
+/** Fetch fruit facts and add them to the DOM. */
+function getFruitFacts() {
+  fetch('/data').then(response => response.json()).then((facts) => {  
+    // Reference each element in facts to create HTML content
+    const factsListElement = document.getElementById('fruit-facts-container');
+    factsListElement.innerHTML = '';
+    for (var factIndex = 0; factIndex < facts.length; factIndex++) {
+      factsListElement.appendChild(
+        createListElement('Fact ' + (factIndex + 1).toString() + ' : ' + facts[factIndex]));
```

Comment:
super nit: maybe let's declare a variable with the text string and then just add that variable to the createListElement. It helps slightly with readability.

Reply:
done!